### DISCUSSION TITLE
what is the name of this button/manager on grapes?

### DISCUSSION THREAD

**Discussion by censonalvin** [GrapesJS/grapesjs#4625] [Q&A]
I was trying to find where or what is the name for this button/what manager is this fullscreen button to customize it (see first picture)? because on my customize UI it does not include  the other stuff that I put (see the second image)

*first
![image](https://user-images.githubusercontent.com/17976518/192708553-f4f1300b-8b87-4258-a707-e01c79d70d75.png)

*second
![image](https://user-images.githubusercontent.com/17976518/192710515-fcc51d91-f7e0-4363-9c8b-c56dcf36b18b.png)

**Comment by ronaldohoch** [comment]
Hello!
There is a lot of ways :D

The command itself: https://github.com/artf/grapesjs/blob/dev/src/commands/view/Fullscreen.js
The button being configured: https://github.com/artf/grapesjs-preset-webpage/blob/master/src/panels.ts#L72

And you can search for it using `editor.Panels.getPanel("options")`
![image](https://user-images.githubusercontent.com/2287371/193042213-e8a10836-572b-4a29-a8d9-9daefe2c2608.png)

**Reply by censonalvin** [reply]
do i need to download the fullscreen.js or it is already integrated when i use the cdn link of grapes?? (sorry for this question. im new being a dev and also on using grapesJS).

**Reply by ronaldohoch** [reply]
Hello! Welcome to this world :D

The fullscreen.js is already in grapesjs from cdn, you can run `editor.runCommand("fullscreen")` to see it working.

Let me ask, this part is a new panel? How do you created it?
![image](https://user-images.githubusercontent.com/2287371/193265889-0474b0a7-072c-4cb1-b77a-5d632a89eb5e.png)

**Reply by censonalvin** [reply]
the ID here in the first pic is where they are placed
![image](https://user-images.githubusercontent.com/17976518/193484389-71f92868-7b20-4c3d-b7dd-a4011cff9323.png)


the second pic is what I used to append them to the IDs of the div from the first pic
![image](https://user-images.githubusercontent.com/17976518/193484443-e63148ba-bc07-4380-878c-88358f89ecbd.png)


*I didn't use react/vue/typescript on this.
(I know my code was a mess as of now. trying to learn more on advance javascript)

**Reply by censonalvin** [reply]
the fullscreen command was working but my sidepanel was not included on the fullscreen command. how can I include my sidepanel on the fullscreen command?

**Reply by ronaldohoch** [reply]
Maybe your sidepanel isn't inside the grapesjs, so the fullscreen command will not get the element.

**Reply by censonalvin** [reply]
how can I include it?

**Reply by ronaldohoch** [reply]
Can you provide some code, not the whole code, but just a new sidepanel so we can check?

Base grapesjs demo \/
https://jsfiddle.net/szLp8h4n

**Reply by censonalvin** [reply]
it was something like this. 

*i use your jsfiddle

https://jsfiddle.net/arnvcjxh/

**Reply by ronaldohoch** [reply]
Well, your sidebar isn't inside grapes.

What are you trying to do? A new panel with grapes buttons or the website sidebar.

**Reply by censonalvin** [reply]
my sidebar is where my blocks is rendered. I was trying to make it like the sidebar of wix ( I mean the design ).

**Reply by ronaldohoch** [reply]
So, you should create a panel with Panel api from grapesjs: 
https://grapesjs.com/docs/api/panels.html#panels

**Reply by censonalvin** [reply]
the said button here is my sidepanel buttons right? so i'll put their i.ds here ?
![image](https://user-images.githubusercontent.com/17976518/195223385-01fef3fa-8bb9-4c64-bac1-fb751bd4b4be.png)

**Reply by censonalvin** [reply]
can't still put my sidepanel. i try to do something like this, but still can't figure it out

> const pn = editor.Panels;
   const panelViews2 = pn.addPanel({
      id: 'options'
    });
    panelViews2.get('buttons').add([{
      attributes: {
        title: 'test'
      },
      label: `Site Manager`,
      command: 'smallsidebar',
      id: 'smngr',
      class: 'smallsidebar-content'
    }]);

**Reply by censonalvin** [reply]
it goes here. not on the sidepanel that i made
![image](https://user-images.githubusercontent.com/17976518/196070211-d4c29222-9c44-4192-b7ad-b6a20372fb6e.png)

**Reply by censonalvin** [reply]
somehow i made it but it making a new button. even if it is not on fullscreen. btw it is not in the fullscreen mode (on the image sample)

![image](https://user-images.githubusercontent.com/17976518/196070980-b41264f7-9bb7-46b3-9c01-a3c683014879.png)

**Comment by artf** [comment]
The fullscreen command supports a custom target option, so in your case, you can create a new custom command and update the button's command label.

```js
const fullscreenAppCmd = 'fullscreen-app';
// Create new command for the fullscreen
editor.Commands.add(fullscreenAppCmd, {
 run: editor => editor.runCommand('core:fullscreen', { target: 'body' /* you can use your app container selector */  }),
 stop: editor => editor.stopCommand('core:fullscreen'),
})

// Update current fullscreen button
editor.Panels.getButton('options', 'fullscreen').set('command', fullscreenAppCmd)
```
